File: android-client/app/src/main/java/com/example/awscognito/data/model/AuthModels.kt
package com.example.awscognito.data.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class GoogleAuthRequest(
    @Json(name = "id_token")
    val idToken: String,
    val email: String?,
    @Json(name = "full_name")
    val fullName: String?
)

@JsonClass(generateAdapter = true)
data class AppleAuthRequest(
    @Json(name = "identity_token")
    val identityToken: String,
    @Json(name = "authorization_code")
    val authorizationCode: String,
    val email: String?,
    @Json(name = "full_name")
    val fullName: String?
)

@JsonClass(generateAdapter = true)
data class AuthTokenResponse(
    @Json(name = "id_token")
    val idToken: String,
    @Json(name = "access_token")
    val accessToken: String,
    @Json(name = "refresh_token")
    val refreshToken: String?,
    @Json(name = "expires_in")
    val expiresIn: Int
)

File: ios-client/AWSCognito/AWSCognito/Data/Models/AuthModels.swift
//
//  AuthModels.swift
//  AWSCognito
//

import Foundation

struct GoogleAuthRequest: Encodable {
    let idToken: String
    let email: String?
    let fullName: String?

    enum CodingKeys: String, CodingKey {
        case idToken = "id_token"
        case email
        case fullName = "full_name"
    }
}

struct AppleAuthRequest: Encodable {
    let identityToken: String
    let authorizationCode: String
    let email: String?
    let fullName: String?

    enum CodingKeys: String, CodingKey {
        case identityToken = "identity_token"
        case authorizationCode = "authorization_code"
        case email
        case fullName = "full_name"
    }
}

struct AuthTokenResponse: Decodable {
    let idToken: String
    let accessToken: String
    let refreshToken: String?
    let expiresIn: Int

    enum CodingKeys: String, CodingKey {
        case idToken = "id_token"
        case accessToken = "access_token"
        case refreshToken = "refresh_token"
        case expiresIn = "expires_in"
    }
}
